solvers/python_gd.py

```diff
@@ -13,19 +14,48 @@ def set_objective(self, X, y, lmbd):
     def run(self, n_iter):
         n_samples, n_features = self.X.shape
 
-        L = (np.linalg.norm(self.X) ** 2 / 4) + self.lmbd
+        L = self.compute_lipschitz_cste()
         step = 1. / L
 
         X, y, lmbd = self.X, self.y, self.lmbd
 
         w = np.zeros(n_features)
         for i in range(n_iter):
-            ywTx = y * np.dot(X, w)
+            ywTx = y * (X @ w)
             temp = 1. / (1. + np.exp(ywTx))
-            grad = -np.dot(X.T, (y * temp)) + lmbd * w
+            grad = -(X.T @ (y * temp)) + lmbd * w
             w -= step * grad
 
         self.w = w
 
     def get_result(self):
         return self.w
+
+    def compute_lipschitz_cste(self, max_iter=100):
+        if not sparse.issparse(self.X):
+            return np.linalg.norm(self.X, ord=2) ** 2 / 4 + self.lmbd
+
+        n, m = self.X.shape
```

Comment:
I discovered `scipy.sparse.linalg.svds`  (sparse truncated svd, uses Lanczos method which is faster than Power method)
`L = svds(X, k=1)[1][0] ** 2`